frontend/__tests__/routes/_public+/type-of-application.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { afterEach, describe, expect, it, vi } from 'vitest';
+
+import { action, loader } from '~/routes/$lang+/_public+/apply+/$id+/type-of-application';
+
+vi.mock('~/routes-flow/apply-flow', () => ({
+  getApplyFlow: vi.fn().mockReturnValue({
+    loadState: vi.fn().mockReturnValue({
+      id: '123',
+      state: { applicationDelegate: 'TRUE' },
+    }),
+    saveState: vi.fn().mockReturnValue({
+      headers: {
+        'Set-Cookie': 'some-set-cookie-header',
+      },
+    }),
+  }),
+}));
+
+vi.mock('~/utils/locale-utils.server', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('~/utils/locale-utils.server')>();
+  return {
+    ...actual,
+    getFixedT: vi.fn().mockResolvedValue(vi.fn()),
+  };
+});
```

Comment:
Similar to previous comment, `redirectWithLocale(..)` should also be mocked.

Reply:
Mocked `redirectWithLocale`